content/intro-to-storybook/react/kr/screen.md

```diff
@@ -0,0 +1,180 @@
+
+```javascript
+// src/components/InboxScreen.stories.js
+
+import React from 'react';
+
+import { PureInboxScreen } from './InboxScreen';
+
+export default {
+  component: PureInboxScreen,
+  title: 'InboxScreen',
+};
+
+export const Default = () => <PureInboxScreen />;
+
+export const Error = () => <PureInboxScreen error="Something" />;
+```
+
+비록 `error` 스토리가 제대로 작동할지라도, `TaskList`에 연결할 Redux store가 없기 때문에 `default` 스토리에 문제가 있음을 알 수 있습니다. (또한 단위 테스트로 `PureInboxScreen`을 테스트할 때도 비슷한 문제가 발생할 것입니다.)
+
+![고장난 inbox](/intro-to-storybook/broken-inboxscreen.png)
+
+이 문제를 피하는 한 가지 방법으로 앱의 최상위 수준에서만 컨테이너 컴포넌트를 렌더링 하는 대신 필요한 모든 데이터를 상위의 컴포넌트에서 하위의 컴포넌트로 전달하는 것입니다.
+
+그러나 개발자가 **불가피하게** 컴포넌트 계층의 하위 계층에서 컨테이너를 렌더링 할 필요가 생길 수 있습니다. Storybook에서 전체 또는 대부분의 앱을 렌더링하려면 우리는 이러한 문제에 대한 해결책이 필요합니다.
+
+<div class="aside">
+여담으로 데이터를 하위 계층에 전달하는 것은 합당한 접근 방식입니다. 특히 <a href="http://graphql.org/">GraphQL</a>을 사용하는 경우에 그렇습니다. 저희는 <a href="https://www.chromatic.com">Chromatic</a>을 만들 때 이러한 방법으로 800개 이상의 스토리를 만들었습니다.
+</div>
+
+## 데코레이터와 함께 컨텍스트를 제공하기
+
+좋은 소식은 스토리 내에서 `InboxScreen`에 Redux store를 제공하기는 매우 쉽습니다! 데코레이터(decorators)를 통해 모방된 Redux store를 사용하면 됩니다.
+
+```javascript
+// src/components/InboxScreen.stories.js
+
+import React from 'react';
+import { action } from '@storybook/addon-actions';
+import { Provider } from 'react-redux';
+
+import { PureInboxScreen } from './InboxScreen';
+import { defaultTasksData } from './TaskList.stories';
+
+export default {
+  component: PureInboxScreen,
+  title: 'InboxScreen',
+  decorators: [story => <Provider store={store}>{story()}</Provider>],
+};
+
+// redux store의 매우 간단한 모방
+const store = {
+  getState: () => {
+    return {
+      tasks: defaultTasksData,
+    };
+  },
+  subscribe: () => 0,
+  dispatch: action('dispatch'),
+};
+
+export const Default = () => <PureInboxScreen />;
+
+export const Error = () => <PureInboxScreen error="Something" />;
+```
+
+[Apollo](https://www.npmjs.com/package/apollo-storybook-decorator)와 [Relay](https://github.com/orta/react-storybooks-relay-container) 등 여타 데이터 라이브러리에 대해서도 모방된 컨텍스트를 제공하는 방식은 비슷합니다.
+
+Storybook에서 state를 순환해봄으로써 우리가 올바르게 하고 있는지를 쉽게 테스트할 수 있도록 해줍니다.
+
+<video autoPlay muted playsInline loop >
+
+  <source
+    src="/intro-to-storybook/finished-inboxscreen-states.mp4"
+    type="video/mp4"
+  />
+</video>
+
+## 컴포넌트 기반 개발
+
+우리는 가장 아래에 해당하는 `Task`로부터 시작하여, `TaskList`로 진행하였고 이제 전체 화면을 구성하는 UI를 완성하였습니다. `InboxScreen`은 중첩된 컨테이너 컴포넌트를 수용하고 그에 수반하는 스토리들을 포함하고 있습니다.
+
+<video autoPlay muted playsInline loop style="width:480px; height:auto; margin: 0 auto;">
+  <source
+    src="/intro-to-storybook/component-driven-development-optimized.mp4"
+    type="video/mp4"
+  />
+</video>
+
+[**컴포넌트 기반 개발**](https://blog.hichroma.com/component-driven-development-ce1109d56c8e)은 컴포넌트의 상위 계층으로 올라감에 따른 복잡성을 점진적으로 확장할 수 있도록 해줍니다. 이것의 이점 중 하나는 보다 개발 과정에 집중할 수 있으며 가능한 모든 UI 순열의 적용 범위가 늘어난다는 것입니다. 간단히 말하면, 컴포넌트 기반 개발(CDD)은 더 높은 품질과 복잡성을 가진 사용자 인터페이스를 만들 수 있도록 도와줍니다.
```

Comment:
@katepark1009 when you can, please adjust the link from https://blog.hichroma.com/component-driven-development-ce1109d56c8e to https://www.componentdriven.org/

Reply:
@jonniebigodes All CDD links got updated. Thank you!